ethcore/machine/src/externalities.rs

```diff
@@ -416,6 +416,10 @@ impl<'a, T: 'a, V: 'a, B: 'a> Ext for Externalities<'a, T, V, B>
 		self.env_info
 	}
 
+	fn chain_id(&self) -> u64 {
+		self.machine.params().chain_id
```

Comment:
Just a note that this has the behavior that if EIP-155 is not enabled on a chain, its `CHAIN_ID` will instead return the network id. This is never specified in EIP-1344, but looks like acceptable behavior for now.